### PR TITLE
chore: lightaccount doc improvement

### PR DESCRIPTION
**Pull request by rthomare**
### Focus of this PR:
This PR focuses on updating and adding information related to the Light Account in the documentation.

### Detailed summary:
- Added a link to the LightAccount & Simple Account Deployment Addresses.
- Updated the link to the Light Account Github Repo.
- Added a link to the Quantstamp Audit Report for the Light Account.
- Updated the link to the "Using Your Own" section in the "Smart Accounts" page.
- Updated the content and title of the Light Account page in the documentation.
- Added benchmark results for the gas optimization of Light Account.
- Added information about Light Account being audited and open-sourced.
- Added information about the upcoming Modular 6900 Accounts.
- Updated the link to the "Getting Started" section in the navigation menu.
- Updated the link to the "Introduction" section in the "Smart Accounts" page.
- Updated the link to the "Light Account" section in the "Smart Accounts" page.
- Updated the link to the "Modular Account" section in the "Smart Accounts" page.
- Updated the link to the "Using Your Own" section in the "Smart Accounts" page.
- Updated the link to the "Deployments" section in the "Smart Accounts" page.
- Updated the link to the "Externally Owned Account (EOA)" section in the "Smart Accounts" page.
- Updated the link to the "Using Your Own" section in the "Smart Accounts" page.
- Updated the link to the "SmartAccountProvider" section in the "aa-core" page.
- Updated the link to the "SmartContractAccount" section in the "aa-core" page.
- Updated the link to the "Public Client" section in the "aa-core" page.
- Updated the link to the "LightAccountClient" section in the "aa-alchemy" page.

> The following files were skipped due to too many changes: `site/.vitepress/config.ts`, `site/smart-accounts/accounts/modular-account.md`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by denniswon — DISMISSED** [review]

**Comment by avasisht23** [comment]
make sure to run prettier on the file you edited @rthomare so it passes ci check :)

**Comment by rthomare** [comment]
Wanted @willhennessy to take a look before landing.

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:19]: this paragraph is trying to do too many things at once. Let's structure this doc into discrete sections for clarity
* Overview:  it's important for docs to start at square one. First section should simply define what Light Account is. I think we should keep the section that was here previously, and then add what you wrote as a "Why Light Account" section explaining the benefits. 
* Why Light Account
* * Gas-optimized (benchmarks)
* * Secure (audited, etc)
* * Coming soon: Modular

Copy and voice notes:
* web3 lowercase
* turbocharged is hyperbolic
* 'we've got your back" is a bit too casual / personifying the contract
* "Our benchmark results speak for themselves, giving you the confidence" also too casual. Goal is to follow the [Google style guidelines](https://developers.google.com/style/tone)

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:25]: I think this is accidentally overstating. It almost reads as if it's irresponsible to choose other smart accounts, which isn't true (any audited contract is responsible)

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:27]: the first two sentences sound fluffy. can we be more concrete?

the last sentence
* style: Alchemy is the only word that should be capitalized there
* do we have a family? that's a pretty expansive concept that I'm not sure we're building. Light + Modular are two options but it's not exactly a family. Let's tone this down and be more concrete

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:29]: in my opinion this doc is getting long (which is good!) and for clarity we should split them up into two separate docs:
* Why Light Account
* Using Light Account

left more details in the [asana ticket](https://app.asana.com/0/1205598834384806/1205608436455254/f).  This doesn't have to be a blocker for this PR, we can do it later if easier

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:35]: this is a catch all section that won't be very useful to a reader, or at least make them do the work to find the info they need. it would be better to cover this in an appropriate section above and delete this section

will leave suggestions of how to restructure this

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:37]: great definition of what it is. let's put in the Overview section

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:39]: Overview

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:41]: these could go in overview or Why Light Account?  or perhaps a "Features" section makes sense if those get long

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:45]: let's put this in the "gas-optimized" section of Why Light Account

**Review by willhennessy — COMMENTED** [review]
thanks rohan!! high level feedback is

- to improve readability, let's organize the information in more tightly defined sections
- aim for tone of voice to match the [Google style guidelines](https://developers.google.com/style/tone)

**Comment by avasisht23** [comment]
Hey following up on this. have all open threads been resolved @willhennessy @rthomare

**Comment by denniswon** [comment]
@rthomare @willhennessy could you please review this doc update for light account?

File: site/smart-accounts/accounts/light-account.md
Comment (rthomare) [inline site/smart-accounts/accounts/light-account.md:31]: "designed to meet your web3 needs", it sounds to high level. Can we be more specific? 

"is a fully functioning web3 account, via smart contracts. It offers the following benefits"

File: site/smart-accounts/accounts/light-account.md
Comment (rthomare) [inline site/smart-accounts/accounts/light-account.md:61]: We can probably shorten this.

"Light Account optimizes for transfers (the most common user flow). Below is a table of Light Account benchmark against other smart account implementations:"

**Comment by denniswon** [comment]
@willhennessy @rthomare updated the doc. Note that I went through the entire doc and made the formatting of our doc site consistent and concise in terms of when to use collapsed mode or not for each section. Overall, I made the doc site more concise and less verbose.

Personally, highly prefer less verbosity and conciseness on docs.

File: site/.vitepress/config.ts
Comment (avasisht23) [reply site/.vitepress/config.ts:112]: lmk if you feel otw. agree it's a bit lengthy

File: site/smart-accounts/accounts/choosing-a-smart-account.md
Comment (avasisht23) [inline site/smart-accounts/accounts/choosing-a-smart-account.md:47]:
```suggestion
If Light Account doesn't fit your specific needs, you can always use your own smart account implementation with Account Kit. To learn how, see our guide in the [Using Your Own](/smart-accounts/accounts/using-your-own) section.
```

File: site/smart-accounts/accounts/modular-account.md
Comment (rthomare) [inline site/smart-accounts/accounts/modular-account.md:73]: Hmm this seems out of scope of light account? Why include this removal in this PR? Am I missing context?

File: site/smart-accounts/accounts/modular-account.md
Comment (rthomare) [inline site/smart-accounts/accounts/modular-account.md:18]: Hmm this files seems out of scope of light account? Why include this removal in this PR? Am I missing context?

File: site/.vitepress/config.ts
Comment (rthomare) [inline site/.vitepress/config.ts:45]: In general always good to break up these "new" things into a seperate PR. that way we can rollback faster, or unblock the light account docs faster. But thanks for doing this!!

**Review by rthomare — COMMENTED** [review]
LGTM but I can't approve since I'm the author. lol

File: site/smart-accounts/accounts/light-account.md
Comment (avasisht23) [inline site/smart-accounts/accounts/light-account.md:23]: i'd keep this tbh.

File: site/smart-accounts/accounts/light-account.md
Comment (avasisht23) [inline site/smart-accounts/accounts/light-account.md:25]: i'd also keep this?

**Review by avasisht23 — COMMENTED** [review]
unsure if this needed a rebase, but left a few comments on the changes. 

the current set up of the docs (e.g. intro/overview pages vs. links directly on the section header) is actually based on other feedback. let's revert.

File: site/.vitepress/config.ts
Comment (moldy530) [reply site/.vitepress/config.ts:112]: we remove it from the sidebar because it was too long. Instead `.. Integration Guide` should go in the meta tags for SEO

**Comment by denniswon** [comment]
separated the PR with https://github.com/alchemyplatform/aa-sdk/pull/215 @rthomare

File: site/.vitepress/config.ts
Comment (denniswon) [reply site/.vitepress/config.ts]: help me understand @avasisht23  why are we performing SEO on sidebars not the title/content?

File: site/smart-accounts/accounts/light-account.md
Comment (denniswon) [reply site/smart-accounts/accounts/light-account.md:23]: it's not gone. down in ## Overview section, first sentence

File: site/smart-accounts/accounts/light-account.md
Comment (denniswon) [reply site/smart-accounts/accounts/light-account.md:25]: I feel like people don't really care that the light account is an improvement to SimpleAccount. What's important is that it is efficient, secure, and compatible/portable.

Even web3 devs, most of them don't know about the simple account reference impl by Ethereum, and even those that do know that it is a ref impl, not for production.

File: site/.vitepress/config.ts
Comment (avasisht23) [reply site/.vitepress/config.ts]: yeah unsure. let's remove it from side bar, but add it to seo metatags

Comment (avasisht23) [reply site/.vitepress/config.ts]: saw you split it to a new PR. will note it there too

File: site/smart-accounts/accounts/light-account.md
Comment (avasisht23) [reply site/smart-accounts/accounts/light-account.md:25]: sgtm

**Review by avasisht23 — APPROVED** [review]

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:23]: is it extendable?  it's not eip-6900 compatible so what do we mean by this?

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [reply site/smart-accounts/accounts/light-account.md:25]: good point, I see both sides here.  I lean toward keeping it for the next ~ 2 months because
* many current AA devs know about SimpleAccount and that is their point of reference
* this builds credibility for us -- we're based on the EF's official version and we improved upon it

i think by early next year this becomes less relevant to the audience and we've built up our brand as a contract developer, then we can remove this

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:27]:
```suggestion
Light Account is an [ERC-4337](https://eips.ethereum.org/EIPS/eip-4337) smart account. It's fully production-ready with a security [audit](https://github.com/alchemyplatform/light-account/blob/main/Quantstamp-Audit.pdf), gas optimizations, and [ERC-1271](https://eips.ethereum.org/EIPS/eip-1271) signature support. Additionally, Light Account support ownership transfer to ensure you and your user don't get locked into a particular signer.
```

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:33]:
```suggestion
Light Account is engineered to minimize gas costs, as measured by the following benchmarks for common operations like native token transfers, ERC20 transfers, and account creation. 
```

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:47]:
```suggestion
Light Account has been audited by Quantstamp. You can find the audit report [here](https://github.com/alchemyplatform/light-account/blob/main/Quantstamp-Audit.pdf). 

Light Account is is also fully open source so you can validate the [source code](https://github.com/alchemyplatform/light-account).
```

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:45]:
```suggestion
### Secure, Audited, Open Source
```

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:51]:
```suggestion
We're pioneering the future of modular smart accounts with [EIP-6900](https://www.alchemy.com/blog/account-abstraction-erc-6900). We're developing a Modular Account to support EIP-6900 coming soon.

Light Account is designed with that future in mind. It is forward-compatible with EIP-6900 so you can upgrade it to 6900-compatible account once the Modular Account is ready.
```

Comment (willhennessy) [reply site/smart-accounts/accounts/light-account.md:51]: also i wasn't sure how to do this, but can you please link the first instance of 'Modular Account' in this comment to the modular account doc in our repo here?

File: site/smart-accounts/accounts/light-account.md
Comment (willhennessy) [inline site/smart-accounts/accounts/light-account.md:53]: feels like this should be pulled out into a separate doc. wdyt? feels buried this far down after all the introduction information

File: site/smart-accounts/accounts/light-account.md
Comment (denniswon) [reply site/smart-accounts/accounts/light-account.md:23]: meaning forward-compatible with modular account via future upgrade

Comment (avasisht23) [reply site/smart-accounts/accounts/light-account.md:23]: ah so maybe the keyword was upgradeable?

Comment (denniswon) [reply site/smart-accounts/accounts/light-account.md:23]: yeah, but upgradeable kind overlapped with the general upgradeable contracts concept, but that's not really what I tried to entail. More of high level umbrella term to imply the overall picture. No strong pref

File: site/smart-accounts/accounts/light-account.md
Comment (denniswon) [reply site/smart-accounts/accounts/light-account.md:53]: We already have a separate light account doc section that goes in depth.

**Pull request merged**
